### DISCUSSION TITLE
conference interpreter mode

### DISCUSSION THREAD

**Discussion by marcellen** [niedev/RTranslator#67] [Ideas]
It would be great to have a real-time "interpreter" mode that provides real-time translation during a conference. This feature should include the following capabilities:

- Choose the audio source: either A) an ambient microphone for live conferences, or B) embedded audio output for online conferences.
- Select the known language in advance to bypass language detection, allowing for faster translations.
- Display the detected audio transcript in both the original and translated languages.
- Real-time
- Save the full transcript for later use.

During a 30 minutes or 1 hour conference, is not practical to press a reconding button, upload audio and have it translated. All work should happen in real time with a reasonable time delay.

**Comment by niedev** [comment]
Hi, thank you for the suggestions, the problem is that with a WalkieTalkie-like mode a real-time translation would be practically impossible, because the speech recognition has to stop when the TTS speaks (both because it would be captured by the microphone and because it would interrupt the speaker).

The mode that comes closest to a real-time translation is the Conversation mode with a bluetooth headset connected (it is not exactly real-time but the interlocutors can speak and hear the translation as if it were), although that can only be used for live conferences and with a maximum of 4 phones connected (to make it work with multiple phones and for a remote conference I would have to have a server, but this would involve costs so this feature should have a subscription fee, and for now I prefer to make everything run locally and for free).

However some of the feature you mentioned will definitely be implemented in the future (like the option to save the transcriptions and the option to display the transcription and the translation in WalkieTalkie chat).

**Comment by anuau** [comment]
Following up on the request for a conference interpreter mode. One-way real-time audio interpretation would be enough for a start.

**Comment by niedev** [comment]
Hi @anuau, first of all, thank you for your donation!
As I said in the comment above, a one device real time translation for more than 2 people (and languages) is impossible. But probably in the future I will extend the number of users that WalkieTalkie and Conversation modes can support. And certainly I will improve the speed and quality of speech recognition and translation 🚀
If I misunderstood you request let me know and explain it in more details.

**Comment by anuau** [comment]
Hi! What I meant was a one-way translation, for example, to be used during
a lecture as a student to understand what the lecturer is saying. I’ve done
extensive research, and the only options I’ve found are TransGull, PolyPal,
and Felo Translator, but they are very expensive.

**Comment by niedev** [comment]
Ok, now I get it, this would be very useful. A simultaneous version of this would still be very difficult and not possible in the short term. But I can implement an audio queue on the WalkieTalkie mode to keep listening even during the audio processing. In that way, if you mute the TTS, in practice it will do the same thing (splitting the transcription in messages, but without missing anything), plus the WalkieTalkie mode will be improved in general. I added this to my list of improvements to do 🚀 Thank you for your suggestion!

**Comment by marcellen** [comment]
I used some other apps meanwhile, like Voiceping and 3PO. 
But only the first one has Windows version too. Still far from perfection on a desktop computer, but on mobile they are quite good.

**Comment by anuau** [comment]
From my experience, when it comes to an audio queue that allows continued
listening during audio processing, the best results occur when single
sentences (around 10 words long) are processed at a time. Some (commercial)
apps process entire paragraphs at once, but this is not the best approach,
as it creates a significant delay between what is being said and what the
user hears.

**Comment by niedev** [comment]
@anuau no problem, the length of the audio processed will still be controllable indirectly by changing the end voice timeout in RTranslator settings. So the user can chose if they want longer or shorter audio split 🚀

**Comment by anuau** [comment]
Thanks. That's a great idea. Don't feel pressured to implement it. For the
time being, I'll try using two phones in conversation mode.
